MobiFlight/Joysticks/Joystick.cs

```diff
@@ -159,11 +167,18 @@ protected string GetFriendlyButtonName(string name, int v)
             return MapDeviceNameToLabel(Regex.Replace(name, @"\d+", v.ToString()).ToString());
         }
 
-        public virtual string MapDeviceNameToLabel(string deviceName)
+        public string MapDeviceNameToLabel(string deviceName)
         {
-            var result = deviceName;
+            // First try and look for a custom label.
+            var input = definition?.FindInputByName(deviceName);
+            if (input != null)
+            {
+                return input.Label;
+            }
```

Comment:
Custom label lookup that used to happen in derived classes can now happen here. If one isn't found then the previous logic is used.